ast2vec/repo2/base.py

```diff
@@ -47,72 +42,27 @@ def __init__(self, tempdir=None, linguist=None, log_level=logging.INFO,
-        if self._linguist is None:
-            self._linguist = shutil.which("enry", path=os.getcwd())
-        if self._linguist is None:
-            self._linguist = "enry"
-        full_path = shutil.which(self._linguist)
-        if not full_path:
-            raise FileNotFoundError("%s was not found. Install it: python3 -m ast2vec enry" %
-                                    self._linguist)
-        with open(full_path, "rb") as fin:
-            self._is_enry = fin.read(15) != b"#!/usr/bin/ruby"
+        self._cloner = RepoCloner(linguist=linguist, redownload=True, log_level=log_level)
         self._bblfsh = [BblfshClient(bblfsh_endpoint or "0.0.0.0:9432")
                         for _ in range(multiprocessing.cpu_count())]
         self._timeout = timeout
 
-    def clone_repository(self, url, save_dir=None):
-        """
-        Clones repository from provided url and saves it to directory target_dir.
-        if target_dir is not provided it saves to temp folder and return it name
-
-        :param url: a URL to clone
-        :param save_dir: path to save cloned repo
-        :return: target_dir or temp folder path
-        """
-        if save_dir is None:
-            target_dir = tempfile.mkdtemp(prefix="repo2base-", dir=self._tempdir)
-        else:
-            target_dir = save_dir
-        url = type(self).prepare_reponame(url)
-        env = os.environ.copy()
-        env["GIT_TERMINAL_PROMPT"] = "0"
-        self._log.info("Cloning from %s...", url)
-        try:
-            subprocess.check_output(
-                ["git", "clone", "--depth=1", url, target_dir],
-                env=env, stdin=subprocess.DEVNULL, stderr=subprocess.STDOUT)
-        except subprocess.CalledProcessError as e:
-            if save_dir is None:
-                shutil.rmtree(target_dir, ignore_errors=True)
-            self._log.error("Git failed to clone repo. git stderr:\n\t" +
-                            "\n\t".join(e.output.decode("utf8").split("\n")))
-            raise e from None
-        except Exception as e:
-            if save_dir is None:
-                shutil.rmtree(target_dir, ignore_errors=True)
-            self._log.error("Unknown error in %s.clone_repository()." % type(self).__name__ +
-                            " Git failed to clone repo.")
-            raise e from None
-
-        return target_dir
-
     def convert_repository(self, url_or_path):
         """
         Queries bblfsh for the UASTs and produces smth useful from them.
 
-        :param url_or_path: Fiel system path to the repository or a URL to clone.
+        :param url_or_path: File system path to the repository or a URL to clone.
         :return: Some object(s) which are returned from convert_uasts().
         """
         temp = not os.path.exists(url_or_path)
         if temp:
-            target_dir = self.clone_repository(url_or_path)
+            target_dir = tempfile.mkdtemp(
+                prefix="repo2nbow-", dir=self._tempdir)
```

Comment:
may be it is time to rename `repo2nbow` prefix to `repo2` prefix because it is common